hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -457,16 +507,32 @@ public ObjectNamespace getObjectNamespace() {
     }
 
     public Map<String, IndexConfig> getIndexDefinitions() {
+        return isGlobalIndexEnabled()
+                ? getGlobalIndexDefinitions()
+                : getPartitionedIndexDefinitions();
+    }
+
+    private Map<String, IndexConfig> getGlobalIndexDefinitions() {
         Map<String, IndexConfig> definitions = new HashMap<>();
-        if (isGlobalIndexEnabled()) {
-            for (Index index : globalIndexes.getIndexes()) {
-                definitions.put(index.getName(), index.getConfig());
+        InternalIndex[] indexes = globalIndexRegistry.getIndexes();
+        for (int i = 0; i < indexes.length; i++) {
+            definitions.put(indexes[i].getName(), indexes[i].getConfig());
+        }
+        return definitions;
+    }
+
+    private Map<String, IndexConfig> getPartitionedIndexDefinitions() {
```

Comment:
It seems from the definition point of view the indexes are the same. Do we need a split global vs local here?

Reply:
maybe no need, didn't change anything here, only refactoring to easily see this kind of things.
Is it ok to consider this change in a separate PR?